src/hooks.rs

```diff
@@ -0,0 +1,264 @@
+use std::ffi::CStr;
```

Comment:
I am okay if you then wrap every single `unsafe fn`'s internals in a single giant `unsafe {}` block with a brief comment on what's going on that makes it internally `unsafe` as well, but I would prefer to be able to see where the `unsafe {}` is. If any of the existing `unsafe fn` are not actually `unsafe`, then I would like to see that too.
```suggestion
#![deny(unsafe_op_in_unsafe_fn)]
use std::ffi::CStr;
```

Reply:
I don't follow?  Is `std::ffi::CStr` some kind of problem?

Reply:
We talked about this offline and I understand.  Will do.